File: code/c/control-flow/test-while.c
/* Program: test-while.c */
#include <stdio.h>
#include <stdbool.h>

bool beg_for_mercy()
{
    char ch;

    printf("Beg for mercy? [y/N]: ");
    scanf("%c", &ch);
    
    return ch == 'y' || ch == 'Y';
}

int main()
{
    bool mercy = false;
    
    printf("Before you stands a 12 foot tall Knight...\n");
    printf("\"We are the Knights who say 'Ni'.\"\n");
    printf("\"I will say Ni to you again if you do not appease us!\"\n");
    
    mercy = beg_for_mercy();
    
    while( !mercy )
    {
        printf("\"Ni!\"\n");
        mercy = beg_for_mercy();
    }
    
    printf("\"Bring us a Shrubbery!\"\n");
    
    return 0;
}

File: code/pascal/control-flow/TestWhile.pas
program TestWhile;

function BegForMercy(): Boolean;
var
    ch: Char;
begin
    Write('Beg for mercy? [y/N]: ');
    ReadLn(ch);
    
    result := (ch = 'y') or (ch = 'Y');
end;

procedure Main();
var
    mercy: Boolean = False;
begin
    WriteLn('Before you stands a 12 foot tall Knight...');
    WriteLn('"We are the Knights who say ''Ni''."');
    WriteLn('"I will say Ni to you again if you do not appease us!"');
    
    mercy := BegForMercy();
    
    while not mercy do
    begin
        WriteLn('"Ni!"');
        mercy := BegForMercy();
    end;
    
    WriteLn('"Bring us a Shrubbery!"');
end;

begin
    Main();
end.